tests/device_tests/test_msg_resetdevice_supershamir.py

```diff
@@ -0,0 +1,232 @@
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),
+                    proto.ButtonRequest(code=B.Other),
+                    proto.ButtonRequest(code=B.Success),  # show seeds ends here
+                    proto.ButtonRequest(code=B.Success),
+                    proto.Success(),
+                    proto.Features(),
+                ]
+            )
+            self.client.set_input_flow(input_flow)
+
+            # No PIN, no passphrase, don't display random
+            device.reset(
+                self.client,
+                display_random=False,
+                strength=strength,
+                passphrase_protection=False,
+                pin_protection=False,
+                label="test",
+                language="english",
+                backup_type=ResetDeviceBackupType.Slip39_Multiple_Groups,
+            )
+
+        # Check if device is properly initialized
+        resp = self.client.call_raw(proto.Initialize())
+        assert resp.initialized is True
+        assert resp.needs_backup is False
+        assert resp.pin_protection is False
+        assert resp.passphrase_protection is False
+
+
+def validate_mnemonics(mnemonics, threshold, expected_ems):
+    # 3of5 shares 3of5 groups
+    # TODO: test all possible group+share combinations?
```

Comment:
That'd be nice :). Let's do it or note it somewhere (new issue? or one larger issue for everything).

Reply:
The only issue I see with this is that if we try to test all combinations it might add a lot of time to the test. It should probably be part of the larger issue.